### PR TITLE
Add process api input buffer helpers

### PR DESCRIPTION
**Pull request by aretecode**

### CODE REVIEW

**Comment by aretecode** [comment]
https://github.com/noflo/noflo/issues/452

File: spec/Component.coffee [hunk 1]
```diff
@@ -1249,3 +1249,85 @@ describe 'Component', ->
...
+        c = new component.Component
+          inPorts:
+            in:
+              datatype: 'string'
+          outPorts:
+            out:
+              datatype: 'string'
+          process: (input, output) ->
+            return unless input.has 'in', (ip) -> ip.type is 'data'
+            data = input.buffer.get 'in', (ip) -> ip.type is 'data'
```
Comment (trustmaster) [inline spec/Component.coffee]: `data` got me confused here, `buffer` or `buf` is a more clear name for what it is.

File: src/lib/Component.coffee [hunk 2]
```diff
@@ -192,6 +193,45 @@ class ProcessInput
...
+    b.filter cb
+
+  # Find packets and modify the original buffer
+  # cb is a function with 2 arguments (ip, index)
+  filter: (name, cb) ->
+    if typeof name is 'function'
+      cb = name
+      name = null
+
+    if @context.scope isnt null
```
Comment (trustmaster) [inline src/lib/Component.coffee]: Can't this just be

```coffeescript
b = @get name
b = @find name, cb
```

also returning the modified buffer here makes sense

**Comment by trustmaster** [comment]
@aretecode a few minor notes, as for the rest a very neat PR.

**Comment by trustmaster** [comment]
@aretecode great job!
@bergie this is good to merge

**Pull request merged**